lib/Doctrine/DBAL/Cache/ResultCacheStatement.php

```diff
@@ -210,4 +312,51 @@ public function rowCount()
 
         return $this->statement->rowCount();
     }
+
+    /**
+     * @return array<string,mixed>|false
+     *
+     * @throws DriverException
+     */
+    private function doFetch()
+    {
+        if ($this->data === null) {
+            $this->data = [];
+        }
+
+        if ($this->statement instanceof ForwardCompatibleResultStatement) {
+            $row = $this->statement->fetchAssociative();
+        } else {
+            $row = $this->statement->fetch(FetchMode::ASSOCIATIVE);
+        }
+
+        if ($row !== false) {
+            $this->data[] = $row;
+
+            return $row;
+        }
+
+        $this->emptied = true;
+
+        return false;
+    }
+
+    /**
+     * @param  array<int,array<mixed>> $data
+     *
+     * @return array<int,array<mixed>>
+     */
+    private function store(array $data, bool $isArray) : array
```

Comment:
:thinking: what's the point of that second argument? It's always `true`, and the first argument is declared as `array` in the signature

Reply:
It was needed before the introduction of `FetchUtils` to implement the "column" fetch modes. It's no longer needed, I’ll remove it.